### DISCUSSION TITLE
Cypress browser tab renders css differently than non-Cypress controlled tab

### DISCUSSION THREAD

**Discussion by reposman33** [cypress-io/cypress#17844] [Questions and Help]
Hello,

I have a login page. It displays a username/ pw form and an empty navbar on top. When I succesfully authenticate by entering username / pw the navigation buttons and a Log out button are rendered in the navbar.

When I manually log in it's like this:

![image](https://user-images.githubusercontent.com/3200870/130357456-3ee57ed9-cf9e-4345-9398-ac698680629f.png) | ![image](https://user-images.githubusercontent.com/3200870/130357486-2dc38a97-bc46-4078-b8c7-1e20aadb052d.png) |
![image](https://user-images.githubusercontent.com/3200870/130357521-ce1c4615-6c7a-4453-8059-ea4cd4d319fd.png)

Notice that after succesfully signing in, the buttons in the navbar are visible. I wrote a Cypress test doing just that. It fills in the username / pw, clicks the "Log in" button. Then it checks if the buttons are visible.  But the navbar stays empty and the test fails. 

The error message Cypress shows (after the built in 40000 ms delay):

>Timed out retrying after 4000ms: expected '<button.button.is-danger.has-text-weight-bold>' to be 'visible'
>
>This element <button.button.is-danger.has-text-weight-bold> is not visible because its parent <div.navbar-menu> has >CSS property: display: none

<button.button.is-danger.has-text-weight-bold> refers to the Log out button. It is the first in the list of buttons to test the visibility in the Cypress test. When I remove that button from the test and re-run the test, the next button triggers this error message.

I expect the navigation buttons and the Log out button to appear, as happens when I manually log in.

This is my html of the navbar

```<div class="navbar shadow-grey-medium is-info" role="navbar">
	<div class="navbar-menu" *ngIf="isAuthenticated">
		<div class="navbar-start">
			<div class="space mr-6"></div>

			<div class="navbar-item">
				<button
					class="button is-light is-outlined"
					routerLink="photos"
					data-cy="Toon Photos"
				>
					Toon Photos
				</button>
			</div>
			<div class="navbar-item">
				<button
					class="button is-light is-outlined ml-4"
					routerLink="albums"
					data-cy="Toon Albums"
				>
					Toon Albums
				</button>
			</div>
			<div class="navbar-item">
				<button
					class="button is-light is-outlined mx-4"
					routerLink="users"
					data-cy="Toon Users"
				>
					Toon Users
				</button>
			</div>
			<div class="navbar-item">
				<button
					class="button is-light is-outlined mr-4"
					routerLink="posts"
					data-cy="Toon Posts"
				>
					Toon Posts
				</button>
			</div>
		</div>
		<div class="navbar-end">
			<div class="navbar-item">
				<button
					class="button is-danger has-text-weight-bold"
					(click)="signOut()"
					data-cy="Log out"
				>
					Log out
				</button>
			</div>
		</div>
	</div>
</div>
```

this is the Cypress test:
```
describe('test login', () => {
	it('Log in Angular App', () => {
		cy.visit('http://localhost:4200')
		cy.login('test@user.net', '******')
		cy.get("[data-cy='Log out']").should('be.visible')
		cy.get("[data-cy='Toon Photos']").should('be.visible')
		cy.get("[data-cy='Toon Albums']").should('be.visible')
		cy.get("[data-cy='Toon Users']").should('be.visible')
		cy.get("[data-cy='Toon Posts']").should('be.visible')
	})
})
```

When I inspect the html using developer tools I see the property `display: none` on the `div.navbar-menu` element containing the buttons. This property is also present when I manually log in in a browser window but its children (div.menu-start and div.menu-end) have a property `display: flex`, overruling the parent's `display: none` and effectively showing the navigation and logout buttons.

Below the relevant html in the browser tab controlled by Cypress after the test is run:

![image](https://user-images.githubusercontent.com/3200870/130358192-1b46e135-ee07-4f44-99ab-e70a0e898173.png)

Notice the css property display:none and the greyed-out children containing the buttons, meaning their css display property is not set.

Below the html rendered in Cypress controlled tab:

![image](https://user-images.githubusercontent.com/3200870/130357193-8f5552b7-d785-4547-a16a-8e7a1731dc83.png)
:

The navbar is empty.

Below the same html snippet after I manually logged in:

![image](https://user-images.githubusercontent.com/3200870/130358324-c15d9cd7-06ec-4dbd-bcc8-00abb3948409.png)

Notice the css property `display:none` but the children containing the buttons are not greyed-out, meaning their css display property is not `none`.

html rendered:

![image](https://user-images.githubusercontent.com/3200870/130357249-d2568605-732e-44b3-93c4-807d4677cf90.png)

notice the buttons in the navigation bar

How can I have Cypress see these DOM elements and have the test succeed?

Cypress v8.3.0
Using the Bulma css framework v 0.9.3
Built with Angular 12.1.4
Firefox 91 used by Cypress and me

**Comment by portyu9** [comment]
Hi @reposman33 

This is a viewport-size issue, not Cypress rendering CSS differently.

Your Cypress screenshot shows the AUT viewport is:

```text
1000 x 660
```

but Bulma applies the desktop `.navbar-menu` styles at:

```css
@media screen and (min-width: 1024px)
```

At 1000px, `.navbar-menu` therefore remains `display: none`.

Set the Cypress viewport above the Bulma breakpoint:

```js
cy.viewport(1280, 720)
```

or globally:

```js
module.exports = {
  viewportWidth: 1280,
  viewportHeight: 720,
}
```

Once the viewport is >= 1024px, the same desktop navbar CSS should apply as in your normal browser.

Please mark this answer as accepted if it helped.
